### DISCUSSION TITLE
modern string functions

### DISCUSSION THREAD

**Discussion by connorhu** [FriendsOfSymfony1/symfony1#343] [Ideas]
We should start using this functions:

- str_contains
- str_starts_with
- str_ends_with

`php-cs-fixer` can do this migration

on php7.4 we should use `symfony/polyfill-php80`

**Comment by alquerci** [comment]
What kind of improvement allow its functions?

- Execution time?
- Readability?
- Security?

**Reply by connorhu** [reply]
Readability.

**Reply by connorhu** [reply]
I've made an experiment.
str_starts_with function: `5.31s user 3.11s system 95% cpu 8.847 total`
prefix check with with subsrt function: `5.50s user 3.30s system 95% cpu 9.218 total`

It's not the time of the run that's interesting, it's the rate.

```php
<?php

if (str_starts_with('string', 'str') === true) {
    true;
}
```

vs

```php
<?php

if (substr('string', 0, strlen('str')) === 'str') {
    true;
}
```

So we can say that the running performance will also be better on modern versions.

**Comment by thePanz** [comment]
Good idea, can you open a PR?
Ideally Doctrine1 should also be "fixed" :+1:
